src/hooks/system_mod.rs

```diff
@@ -0,0 +1,66 @@
+use rhai::{Dynamic, EvalAltResult, Module};
+use std::process::Command;
+
+use crate::{
+    interactive::prompt_for_variable,
+    project_variables::{StringEntry, TemplateSlots, VarInfo},
+    Args,
+};
+
+type Result<T> = anyhow::Result<T, Box<EvalAltResult>>;
+
+/// Creates the system module, containing the `command` function,
+/// which allows you to run system command.
+pub fn create_module(args: &Args) -> Module {
+    let mut module = Module::new();
+
+    let allow_commands = args.allow_commands;
+    module.set_native_fn("command", move |name: &str, commands_args: rhai::Array| {
+        run_command(name, commands_args, allow_commands)
+    });
+
+    module
+}
+
+fn run_command(name: &str, args: rhai::Array, allow_commands: bool) -> Result<Dynamic> {
+    let args: Vec<String> = args.into_iter().map(|arg| arg.to_string()).collect();
+
+    // If the user specified the --allow-commands flag, don't prompt.
+    let should_run = allow_commands || {
+        let prompt = format!(
+            "The template is requesting to run the following command. Do you agree?\n{} {}",
+            name,
+            args.join(" ")
+        );
+
+        // Prompt the user for whether they actually want to run the command.
+        let value = prompt_for_variable(&TemplateSlots {
+            prompt,
+            var_name: "".into(),
+            var_info: VarInfo::String {
+                entry: Box::new(StringEntry {
+                    default: Some("no".into()),
+                    choices: Some(vec!["yes".into(), "no".into()]),
+                    regex: None,
+                }),
+            },
+        });
+
+        // Only accept clearly positive affirmations.
+        matches!(
+            value.map(|s| s.trim().to_ascii_lowercase()).as_deref(),
+            Ok("yes" | "y")
+        )
+    };
+
+    if !should_run {
+        return Ok(Dynamic::UNIT);
+    }
```

Comment:
Also unsure about this - maybe it's better to just panic?

Reply:
No, please return a proper error desribing the reason for the failure: User has denied execution of external command.

`cargo-generate` might be run in 'silent' mode, thus the prompt is disabled and the user never got to see it!

Reply:
>return a proper error desribing the reason for the failure

I'm assuming you mean return from cargo-generate entirely? My thoughts were initially that the command could fail but only return to the Rhai script (the Rhai script could then handle the failing command however it wants) but I think this probably won't be very useful and you could always simulate it by running a bash script as the system command.

I added some handling for the silent mode :)

Reply:
I don't have a hard opinion here - maybe @sassman has?

Reply:
not really, let's see how this flies and if users are running into issues with it.